### PR TITLE
Redact secrets for flagz

### PR DESCRIPTION
**Pull request by tempoz**
Leverages `autoflags.New` to tag flags as being secret so that they can be redacted in flagz.

Refactors `autoflags` so that deprecation and marking secret can easily be used in concert using the tags.

Refactors `flagutil/yaml` so that we can remove entries from documented YAML and so that we can use information from the flag in `DocumentNodeOption`s.

These refactors allow us to implement a new option, `RedactSecrets`, which uses hints from struct tags and the flag, if present, to determine if a field needs to be redacted and remove it if it does.

---

**Version bump**: Minor

### CODE REVIEW

**Comment by tempoz** [comment]
Current dependencies on/for this PR:
* master
  * **PR #2601** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/2601" target="_blank">Graphite</a>  👈
    * **PR #2602** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/2602" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/2601?utm_source=stack-comment).

**Comment by tempoz** [comment]
Tested manually.

**Review by tylerwilliams — APPROVED** [review]

**Comment by tempoz** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/2601?utm_source=stack-comment).


Job Reference: lu0xHwGRTBLZtDhLZ3d5

**Pull request merged**
